contrib/msggen/msggen/schema.json

```diff
@@ -16136,6 +16136,168 @@
         }
       ]
     },
+    "lightning-listaddresses.json": {
+      "$schema": "../rpc-schema-draft.json",
+      "type": "object",
+      "added": "v24.11",
+      "rpc": "listaddresses",
+      "title": "Command to list all addresses issued by the node to date",
+      "description": [
+        "The **listaddresses** RPC command provides a detailed list of all Bitcoin addresses that have been generated and issued by the Core Lightning node up to the current date."
+      ],
+      "categories": [
+        "readonly"
+      ],
+      "request": {
+        "required": [],
+        "additionalProperties": false,
+        "properties": {
+          "start": {
+            "type": "u64",
+            "description": [
+              "Starting index for listing addresses or searching for a particular address."
+            ],
+            "default": 0
+          },
+          "end": {
+            "type": "u64",
+            "description": [
+              "Ending index for listing addresses or searching for a particular address."
+            ],
+            "default": "Total number of addresses issued"
+          },
+          "address": {
+            "type": "string",
+            "description": [
+              "Any Bitcoin accepted type, including bech32."
+            ]
+          },
+          "allow_future_addr": {
```

Comment:
This is not something we want to expose. It's a great way to get users to run into an issue where they use a future address, but the node does not watch for it when scanning blocks.

I'm guessing this is a result of the end parameter being an absolute Index. Making it in to a limit (that can be undershot) removes this issue.

Reply:
Also what happens if this disallows, but the index is still in the future (both indices can be in the future by the way)? Does it error out, or does it silently skip future addresses (in which case why not make that the default and not expose the extra param?)

Reply:
It is used in our test to get future address and I didn't realize that I can use `dev` params too :). Now it will only be allowed in `developer` mode.

If the future addresses were not allowed but the index was in future then we were only returning the list of addresses which are already issued. It `start` and `end` both were future indices then it was returning empty list.